lua/dirbuf/planner.lua

```diff
@@ -189,7 +228,19 @@ function M.determine_plan(changes)
   end
 
   for _, fs_entry in ipairs(changes.new_files) do
-    table.insert(plan, create(fs_entry))
+    local fnames = expand_path(fs_entry.fname)
+    local paths = expand_path(fs_entry.path)
+
+    for i=1,#fnames do
+      table.insert(plan, create({
+        fname = fnames[i],
+        ftype = fs_entry.ftype,
+        path = paths[i]
+      }))
+    end
```

Comment:
Instead of doing expansion here, I think it's better to do it as a pre-processing step [right before calling `planner.build_changes()`](https://github.com/bscuron/dirbuf.nvim/blob/ca90cc8f4dd0b4aeeb9d67b828f84985e6b176b2/lua/dirbuf.lua#L333). That leads to more consistent behavior where the `{a,b,c}` syntax works both for new lines and old lines (and cleaner code too!)
```
#1 a{b,c}
z{x,y}
```
expands to
```
#1 ab
#1 ac
zx
zy
```

(This does break the current line number tracking but that shouldn't be too hard to fix. Also there are edge cases with commas inside of `{}` which need to be handled and we now need to escape `{}`.)

Reply:
> Instead of doing expansion here, I think it's better to do it as a pre-processing step [right before calling `planner.build_changes()`](https://github.com/bscuron/dirbuf.nvim/blob/ca90cc8f4dd0b4aeeb9d67b828f84985e6b176b2/lua/dirbuf.lua#L333). That leads to more consistent behavior where the `{a,b,c}` syntax works both for new lines and old lines (and cleaner code too!)
> 
> ```
> #1 a{b,c}
> z{x,y}
> ```
> 
> expands to
> 
> ```
> #1 ab
> #1 ac
> zx
> zy
> ```
> 
> (This does break the current line number tracking but that shouldn't be too hard to fix)

I also did not run into the line numbering breaking from my testing so far.

Reply:
Sorry, I should have been more clear. When I said line number tracking I was thinking about dirbuf.nvim's error messages.

So once you expand a line, all future error messages will be offset by however many more lines that line expanded to. For example this example will report a `line 4: Duplicate name 'foo'` even tho the duplicate actually occurs on line 3.
```
foo
a{b,c}
foo
```